[Kai Siang] iP

kslui99 opened the pull request:
### Duke
> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use 

All you need to do is,
1. download duke.jar from [here](https://github.com/kslui99/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks 
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
complete level-1: add greet, echo, exit

Commit:
complete level-2: add list, add

Commit:
complete level-3: mark as done, and extension: A-Classes

Commit:
complete level-4: ToDo, Event, Deadline and Extension: A-Inheritance

Commit:
complete A-TextUiTesting: Automated Text UI Testing

Commit:
complete level-5: Handle Errors

Commit:
complete level-6: Delete

Commit:
Change chatbot name and use class-level instead of object-level members

Commit:
Add automatic save

Commit:
Use LocalDate for dates and times

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
Refactor to use more OOP

Commit:
Organize into package

Commit:
Add Junit tests

Commit:
Create a JAR file

Commit:
Add JavaDoc comments to all public classes and methods

Commit:
Tweak code to comply with coding standard

Commit:
Add find

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'add-gradle-support' of https://github.com/kslui99/ip

Commit:
Set up gradle to build and run Duke, and run unit tests

Commit:
Add checkstyle to detect coding style violations

Commit:
Fix coding style violations detected by checkstyle

Commit:
Add GUI using JavaFX

Commit:
Merge branch 'branch-Level-10'

jianh0ng commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,67 @@
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        Task t = null;
+
+        // Create Todo, Deadline or Event task
```
I like the use of comments inside methods to increase understandability.

jianh0ng commented on src/main/java/duke/DoneCommand.java:
```diff
@@ -0,0 +1,42 @@
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        // Handle user inputs that are out of bounds
```
Nice catching of potential exceptions

jianh0ng reviewed (commented):
Very well documented code, good use of OOP and handling of errors. Looks good :)

Commit:
Use Java assertions

kslui99 committed:
Merge pull request #1 from kslui99/branch-A-Assertions

Use Java assertions

Commit:
Merge branch 'master' of https://github.com/kslui99/ip

Commit:
Refactor to improve code quality

There are several instances of code duplication in a switch statement,
and complicated expressions.

Reducing code duplication and simplifying these complicated expressions
improves the code quality.

As a step towards improving the code quality, let's minimize these
code duplication and complicated expressions by refactoring the switch
statements and decomposing the long expressions.

kslui99 committed:
Merge pull request #2 from kslui99/branch-A-CodeQuality

Refactor to improve code quality

Commit:
Merge branch 'master' of https://github.com/kslui99/ip

* 'master' of https://github.com/kslui99/ip:
  Refactor to improve code quality

clementkfj commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,68 @@
+            t = new Deadline(d, by);
+            break;
+        case "event":
+            String s = description.split(" /at")[0];
```
A more meaningful name used to name the String may help improve readability?

clementkfj commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,106 @@
+                String description = splitLine[2];
+
+                // Store data from file into tasks arraylist
+                Task t = null;
```
Suggest to have another name for the task instead of just 't'!

clementkfj commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,106 @@
+            return tasks;
+        } catch (FileNotFoundException e) {
+            // Create the file
+            File f = new File(filePath);
```
Same here, where a new name can be used instead of just f

clementkfj reviewed (commented):
Overall tidy and neat code! Small suggestions that do not harm the program. Great job!

Commit:
Add a way to tag tasks

There is currently no way for the user to add extra notes to a task.

Adding the tag functionality will allow the user to add more information to a task.

As a step towards helping the user categorise the tasks better, let's add a tag functionality so that users can add tags to items.

Commit:
Increase main window height

Commit:
Update README.md, add product screenshot, and fix find command formatting

Commit:
Use more OOP

Commit:
Update README.md

Commit:
Fix exit command to close program window

Commit:
Update README.md

kslui99 committed:
Set theme jekyll-theme-midnight

kslui99 committed:
Set theme jekyll-theme-cayman

Commit:
Run checkstyle and tests and finalize product

Commit:
Merge branch 'master' of https://github.com/kslui99/ip
